pkg/controllers/management/node/cleanup.go

```diff
@@ -153,6 +153,13 @@ func (m *Lifecycle) cleanRKENode(node *v3.Node) error {
 		return nil
 	}
 
+	if _, err := userContext.K8sClient.CoreV1().Nodes().Get(m.ctx, node.Status.NodeName, metav1.GetOptions{}); kerror.IsNotFound(err) {
+		logrus.Debugf("node [%s] is already deleted, cannot clean RKE node", node.Status.NodeName)
+		return nil
+	} else if err != nil {
```

Comment:
out of curiosity, is it possible that (before) we ever got to this point with an unhealthy control plane on the downstream cluster?